File: backend/app/src/main/java/backend/model/unit/UnitResponse.java
package backend.model.unit;

import backend.model.pair.PairResponse;
import backend.model.player.PlayerResponse;

public class UnitResponse {
    public final String id;
    public PairResponse position;
    public int hp;
    public PlayerResponse owner;
    public int damage;
    public int attackSpeed;
    public int movementSpeed;
    public int range;
    public UnitType type;

    public UnitResponse(Unit unit) {
        this.id = unit.id;
        this.position = new PairResponse(unit.position);
        this.hp = unit.hp;
        this.owner = new PlayerResponse(unit.owner);
        this.damage = unit.damage;
        this.attackSpeed = unit.attackSpeed;
        this.movementSpeed = unit.movementSpeed;
        this.range = unit.range;
        this.type = unit.type;
    }
}

File: frontend/src/ts/lib/model/UnitResponse.ts
import PlayerResponse from "./PlayerResponse.js";
import Position from "./Position.js";

export enum UnitType {
  CASTLE = "CASTLE",
  WARRIOR = "WARRIOR",
  ARCHER = "ARCHER",
}

export default class UnitResponse {
  public id: string;
  public position: Position;
  public hp: number;
  public owner: PlayerResponse;
  public range: number;
  public damage: number;
  public attackSpeed: number;
  public movementSpeed: number;
  public type: UnitType;

  constructor(
    id: string,
    range: number,
    hp: number,
    damage: number,
    attackSpeed: number,
    movementSpeed: number,
    position: Position,
    owner: PlayerResponse,
    type: UnitType
  ) {
    this.id = id;
    this.hp = hp;
    this.position = position;
    this.owner = owner;
    this.range = range;
    this.damage = damage;
    this.attackSpeed = attackSpeed;
    this.movementSpeed = movementSpeed;
    this.type = type;
  }
}
